vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+
+```         
+filtered_deals <- reactive({
+      shiny.fluent::fluentSalesDeals |> filter(
+        is_closed | input$includeOpen
+      )
+    })
+```
+
+This code is defining a reactive expression called **`filtered_deals`** in an R Shiny application.
+
+The reactive expression uses the **`|>`** operator to pipe the **`shiny.fluent::fluentSalesDeals`** data frame to the **`filter()`** function which comes from the `dplyr` package.
+The **`filter()`** function is used to filter rows from a data frame based on one or more conditions.
+
+In this case, the **`filter()`** function is filtering rows from **`shiny.fluent::fluentSalesDeals`** based on two conditions:
+
+1.  **`is_closed`**: This is a logical column in the **`shiny.fluent::fluentSalesDeals`** data frame that indicates whether a sales deal has been closed or not.
+
+2.  **`input$includeOpen`**: This is a reactive input object that is defined in the UI part of the code.
+    It is used to determine whether open deals (i.e. deals that have not been closed) should be included in the filtered data or not.
+
+The **`|`** symbol between the two conditions is the logical OR operator, which means that the **`filter()`** function will return all rows where either **`is_closed`** is TRUE or **`input$includeOpen`** is TRUE.
+
+So, the `app/view/datatable.R` file now is modified to -
+
+```         
+box::use(
+  shiny[moduleServer, NS, uiOutput, renderUI, reactive],
+  shiny.fluent[fluentPage, Text, DetailsList, Toggle.shinyInput],
+  tibble[tibble],
+  dplyr[filter]
+)
+
+columns <- tibble(
+    fieldName = c("rep_name", "date", "deal_amount", "city", "is_closed"),
+    name = c("Sales Rep", "Date", "Deal Amount", "City", "Is Closed?"))
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+    fluentPage(
+        Text("A randomly generated dataset of imaginary sales deals"),
+        Toggle.shinyInput(ns("includeOpen"), label = "Include open deals"),
+        uiOutput(ns("datatable")))
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    
+    filtered_deals <- reactive({
+      shiny.fluent::fluentSalesDeals |> filter(
+        is_closed | input$includeOpen
+      )
+    })
+
+    output$datatable <- renderUI({
+        DetailsList(items = filtered_deals(), columns = columns)
+    })
+  })
+}
+```
+
+This is how the app looks now!
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_gif_1.gif")
+```
+
+## 4. Improve visuals for the table with elevated cards
+
+We can create cards with elevations just like in Fluent UI [documentation](https://developer.microsoft.com/en-us/fluentui#/styles/web/elevation).
+Let us style the data table area.
+
+The following code does 2 things -
+
+-   Creates an elevated card around the data table area
+
+-   Confines the height of the table area
+
+```         
+div(
+      class = "ms-depth-64", 
+      style = "height: 500px; overflow: auto;",
```

Comment:
Inline styling is a bad practice! For styling, use Sass setup provided by Rhino.

Reply:
added comment to redirect user to custom styling in rhino documentation